src/data/area/area_picker.ts

```diff
@@ -0,0 +1,199 @@
+          return false;
+        }
+        return deviceEntityLookup[device.id].some((entity) => {
+          const stateObj = haStates[entity.entity_id];
+          if (!stateObj) {
+            return false;
+          }
+          return (
+            stateObj.attributes.device_class &&
+            includeDeviceClasses.includes(stateObj.attributes.device_class)
+          );
+        });
+      });
+      inputEntities = inputEntities!.filter((entity) => {
+        const stateObj = haStates[entity.entity_id];
+        return (
+          stateObj.attributes.device_class &&
+          includeDeviceClasses.includes(stateObj.attributes.device_class)
+        );
+      });
+    }
+
+    if (deviceFilter) {
+      inputDevices = inputDevices!.filter((device) => deviceFilter!(device));
+    }
+
+    if (entityFilter) {
+      inputDevices = inputDevices!.filter((device) => {
+        const devEntities = deviceEntityLookup[device.id];
+        if (!devEntities || !devEntities.length) {
+          return false;
+        }
+        return deviceEntityLookup[device.id].some((entity) => {
+          const stateObj = haStates[entity.entity_id];
+          if (!stateObj) {
+            return false;
+          }
+          return entityFilter(stateObj);
+        });
+      });
+      inputEntities = inputEntities!.filter((entity) => {
+        const stateObj = haStates[entity.entity_id];
+        if (!stateObj) {
+          return false;
+        }
+        return entityFilter!(stateObj);
+      });
+    }
+  }
+
+  let outputAreas = areas;
+
+  let areaIds: string[] | undefined;
+
+  if (inputDevices) {
+    areaIds = inputDevices
+      .filter((device) => device.area_id)
+      .map((device) => device.area_id!);
+  }
+
+  if (inputEntities) {
+    areaIds = (areaIds ?? []).concat(
+      inputEntities
+        .filter((entity) => entity.area_id)
+        .map((entity) => entity.area_id!)
+    );
+  }
+
+  if (areaIds) {
+    outputAreas = outputAreas.filter((area) => areaIds!.includes(area.area_id));
+  }
+
+  if (excludeAreas) {
+    outputAreas = outputAreas.filter(
+      (area) => !excludeAreas!.includes(area.area_id)
+    );
+  }
+
+  const items = outputAreas.map<PickerComboBoxItem>((area) => {
+    const { floor } = getAreaContext(area, haFloors);
+    const floorName = floor ? computeFloorName(floor) : undefined;
+    const areaName = computeAreaName(area);
+    return {
+      id: `${idPrefix}${area.area_id}`,
+      primary: areaName || area.area_id,
+      secondary: floorName,
+      icon: area.icon || undefined,
+      icon_path: area.icon ? undefined : mdiTextureBox,
+      search_labels: {
+        aliases: area.aliases.join(" "),
+      },
+    };
+  });
+
+  return items;
+};
+
+export const areaComboBoxKeys: FuseWeightedKey[] = [
+  {
+    name: "primary",
+    weight: 10,
+  },
+  {
+    name: "search_labels.aliases",
+    weight: 8,
+  },
+  {
+    name: "secondary",
+    weight: 8,
```

Comment:
```suggestion
    weight: 6,
```

So aliases are higher, which may be the same as the floorname. This also matches the weight with what it was.